GsaGH/Helpers/Export/Model/Lists.cs

```diff
@@ -1,3 +1,333 @@
+          .Where(x => x != null && x.Value != null)) {
+          string id = ElementListFromReference.GetReference(member3d.Value.Guid,
+            ReferenceType.MemberChildElements, null, null, null, null, null, apiMembers,
+            memberElementRelationship);
+          if (id == "") {
+            owner.AddRuntimeMessage(GH_RuntimeMessageLevel.Warning,
+              "Issue adding List " + list.Name + " to Model:" + Environment.NewLine
+              + member3d.Value.ToString() + " not found in Model");
+          }
+
+          list._definition += " " + id;
+        }
+      }
+
+      list._definition.Trim();
+
+      AddList(list, ref apiLists);
+    }
+
+    private static void AddMemberList(GsaList list, ref GsaGuidDictionary<EntityList> apiLists,
+      GsaGuidDictionary<Member> apiMembers, GH_Component owner) {
+      if (list._members == (null, null, null) || list._members.m1d.Count == 0
+        || list._members.m2d.Count == 0 || list._members.m3d.Count == 0) {
+        AddList(list, ref apiLists);
+      }
+
+      foreach (GsaMember1dGoo member1d in list._members.m1d
+        .Where(x => x != null && x.Value != null)) {
+        string id = ElementListFromReference.GetReference(member1d.Value.Guid,
+          ReferenceType.Member, null, null, null, null, null, apiMembers, null);
+        if (id == "") {
+          owner.AddRuntimeMessage(GH_RuntimeMessageLevel.Warning,
+            "Issue adding List " + list.Name + " to Model:" + Environment.NewLine
+            + member1d.Value.ToString() + " not found in Model");
+        }
+
+        list._definition += " " + id;
+      }
+
+      foreach (GsaMember2dGoo member2d in list._members.m2d
+        .Where(x => x != null && x.Value != null)) {
+        string id = ElementListFromReference.GetReference(member2d.Value.Guid,
+          ReferenceType.Member, null, null, null, null, null, apiMembers, null);
+        if (id == "") {
+          owner.AddRuntimeMessage(GH_RuntimeMessageLevel.Warning,
+            "Issue adding List " + list.Name + " to Model:" + Environment.NewLine
+            + member2d.Value.ToString() + " not found in Model");
+        }
+
+        list._definition += " " + id;
+      }
+
+      foreach (GsaMember3dGoo member3d in list._members.m3d
+        .Where(x => x != null && x.Value != null)) {
+        string id = ElementListFromReference.GetReference(member3d.Value.Guid,
+          ReferenceType.Member, null, null, null, null, null, apiMembers, null);
+        if (id == "") {
+          owner.AddRuntimeMessage(GH_RuntimeMessageLevel.Warning,
+            "Issue adding List " + list.Name + " to Model:" + Environment.NewLine
+            + member3d.Value.ToString() + " not found in Model");
+        }
+
+        list._definition += " " + id;
+      }
+
+      list._definition.Trim();
+
+      AddList(list, ref apiLists);
+    }
+
+    private static void AddList(GsaList list, ref GsaGuidDictionary<EntityList> apiLists) {
```

Comment:
why ref?

Reply:
the GuidDictionaries are passed around into the different add methods to be updated and checked for existing entries inside each method